app/models/user.server.ts

```diff
@@ -47,6 +47,17 @@ export async function createUser(
   });
 }
 
+export async function deactivateUserByUsername(username: PrismaUser['username']) {
+  return prisma.user.update({
+    where: { username },
+    data: { deactivated: true },
+  });
+}
+
+export async function deleteUserByUsername(username: PrismaUser['username']) {
+  return prisma.user.delete({ where: { username } });
```

Comment:
I implemented this mostly to clean up the database while running the tests. I was thinking this could be called as part of the user delete process. Should I update this to delete the related records for a user as well?

Reply:
What you can do is to add [cascade delete](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions#cascade) in the schema when specifying the relationship.
For example:
`InsurancePolicy    InsurancePolicy    @relation(fields: [policyId], references: [policyId], onDelete: Cascade)`

Reply:
Thanks, I have added this to the `Certificate`, `Record`, and `Challenge` models